src/features/authorSlise.ts

```diff
@@ -0,0 +1,26 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+
+type AuthorState = {
+  author: User | null;
+};
+
+const initialState: AuthorState = {
+  author: null,
+};
+
+const authorSlice = createSlice({
+  name: 'author',
+  initialState,
+  reducers: {
+    setAuthor: (state, action) => {
+      return {
+        ...state,
+        author: action.payload,
+      };
+    },
```

Comment:
```suggestion
    setAuthor: (state, action) => {
    state. author = action.payload,
    },
```
you can safely mutate state in redux toolkit reducer